include/dynamatic/Analysis/ControlDependenceAnalysis.h

```diff
@@ -0,0 +1,82 @@
+//===- ControlDependenceAnalysis.h - Control dependence analyis *--- C++-*-===//
+//
+// Dynamatic is under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Declares some utility functions useful to analyzing the control dependencies
+// between basic blocks of the CFG.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef DYNAMATIC_ANALYSIS_CONTROLDEPENDENCEANALYSIS_H
+#define DYNAMATIC_ANALYSIS_CONTROLDEPENDENCEANALYSIS_H
+
+#include "dynamatic/Support/LLVM.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Pass/AnalysisManager.h"
+#include <optional>
+
+#define DEBUG_TYPE "CONTROL_DEPENDENCE_ANALYSIS"
```

Comment:
This needs to be in the source file, not the header (multiple headers could define it and only the last one to do so would be taken into account from a source file). Also the standard case style would be "control-dependence-analysis".